ui/app/tailwind.css

```diff
@@ -342,18 +342,39 @@
 
 @layer base {
   * {
-    @apply border-border;
+    @apply border-border outline-ring/50;
   }
   body {
     @apply bg-background text-foreground;
   }
 }
 
-@layer base {
-  * {
-    @apply border-border outline-ring/50;
-  }
-  body {
-    @apply bg-background text-foreground;
+@layer components {
+  .CodeBlock {
+    &:where(.CodeBlock--with-line-numbers) code {
+      counter-reset: line;
+    }
+
+    &:where(.CodeBlock--with-line-numbers) .line {
+      counter-increment: line;
+
+      &::before {
```

Comment:
This block is what adds sticky line numbers without creating a dedicated container, since that would require additional custom rendering logic with Shiki if we wanted to use our current approach.